self-core-impl/src/main/java/com/selfxdsd/core/GithubRepo.java

```diff
@@ -56,10 +57,28 @@
 
     @Override
     public Project activate() {
-        return this.storage()
-            .projectManagers()
-            .pick(this.provider())
-            .assign(this);
+        final Project active = super.activate();
+        this.resources().post(
+            URI.create(this.repoUri().toString() + "/hooks"),
+            Json.createObjectBuilder()
+                .add(
+                    "events",
+                    Json.createArrayBuilder()
+                        .add("issues")
+                )
+                .add(
+                    "config",
+                    Json.createObjectBuilder()
+                        .add(
+                            "url",
+                            "https://self-xdsd.com/github/"
+                              + active.repoFullName()
+                        )
+                        .add("content_type", "json")
+                        .add("secret", active.webHookToken())
+                ).build()
+        );
+        return active;
```

Comment:
@criske Cand un repo e "activat", ar trebui sa faca 3 chestii:

1. Sa salveze repo-u un DB, assignat unui PM
2. Sa cheme Github API pt a seta webhook-ul la repository
3. Sa cheme Github API pt a invita PM-ul in Repo.

Cum e facut acum, nu prea imi place, ca metoda asta face toate astea 3 chestii, mi se pare prea mult. 
Idei, sugestii, pareri, sa spargem putin logica asta cumva? :D

Reply:
@amihaiemil Ma gandesc ca am putea sparge logica aia intr-o secventa de micro-taskuri care pot fi testate in izolare unele fata de altele. Rezultatul va fi proiectul activat.
```java
Project project = new StepRunner<Project>()
   .add(new ProjectActivationStep(storage, repo))
   .add(new WebhookRegistrationStep(resources))
   .add(new PmInvitationStep(resources, repo))
   .run()
```
Interfata `Step` ar avea 2 metode:
- `onNext(T data, Stepper<T> stepper)` chemata cand se ajunge la pasul current  unde: 
`data` reprezinta data provenita de la pasul anterior (`Project` in cazul de fata)
`stepper`  o interfata prin care pasul curent invoca pasul urmator.
De ex implementarea `ProjectActivationStep#onNext` ar arata asa:
```java
  this.project = this.storage
            .projectManagers()
            .pick(this.repo.provider())
            .assign(this.repo)
  stepper.next(this.project)
```

- `onError(Exception)` chemata cand se produce o eroare intr-un pas in "downstream". Eroarea
  e propagata catre toti pasi anteriori pasului unde s-a produs eroarea.
  De ex in `WebhookRegistrationStep#onNext` userul nu are drepturi de a seta weebhook-ul atunci va chema
  `stepper.error(new IlegalStateException(...))`. Asta ar rezulta ca `ProjectActivationStep` sa fie notificat 
 prin `onError` iar acolo ma gandesc ca proiectul va fi dezactivat `this.project.deactivate()`. Daca totul e OK 
 `WebhookRegistrationStep`, va pasa mai departe proiectul prin `stepper#next`

Interfata 'Stepper<T>' va avea doua metode:
- `next(T)` cheama pasul urmator
- `error(Exception)` intrerupe flow-ul si notifica toti pasii precedenti de eroare.

Ca tot flow-ul sa functioneze corect fiecare pas trebuie sa cheme in `onNext` fie `stepper#next()` sau `stepper#error()`.

Cat despre implementarea `StepRunner` e destul de simpla: 
https://gist.github.com/criske/c1d482c60cc0ee8d963ff562a7824b2f